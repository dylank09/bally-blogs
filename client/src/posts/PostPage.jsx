import React, { useEffect, useState } from 'react';
import Post from '../components/Post';
import BlogsApi from '../services/BlogsApi';
import { useParams, Link } from 'react-router-dom';

const PostPage = () => {

    const { id } = useParams();
    const [post,setPost] = useState([]);

    useEffect(() => {
        
        const getPost = async () => {
          const postData = await BlogsApi.fetchPost(id);
          setPost(postData)
        }
    
        getPost();
    
      },[id]);

    return (
        <div>
            <nav>
                <Link className="nav-link" to='/posts'>Posts</Link>      <br></br>
                <Link className="nav-link" to='/'>Home</Link>            <br></br>
            </nav>

            <Post 
                id={post.id}
                title={post.title}
                body={post.body}
                date={post.created_at}
            />

        </div>
             
    );
};

export default PostPage;