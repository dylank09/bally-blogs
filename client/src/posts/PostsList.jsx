import React, {useEffect, useState} from 'react';
import Post from '../components/Post';

const PostsList = ({blogApi}) => {

    const [state, setState] = useState({loading: true});
    
    const truncate = (text) => {
        if(text.length > 300) {
            text = text.substring(0, 300);
            text += "...";
            return text;
        }
        else return text;
    }

    useEffect(() => {
        const getPosts = async () => {
          const posts = await blogApi.fetchPosts();
          setState({
            loading: false,
            posts
          });
        }
    
        getPosts();
    
    },[blogApi]);


    if (state.loading) {
        return <div><h2 className="loading">LOADING...</h2></div>
    }
    else {
        return (
            <div>
                {state.posts.length > 0 ? 
                    <span>
                        <h2>Blogs</h2>
                        <ul className="posts-list"> 
                            {state.posts.map(post =>  (
                            
                                <Post key={post.id}
                                    id={post.id}
                                    title={post.title}
                                    body={truncate(post.body)}
                                    date={post.created_at.substring(0,10)}/>
                            ))}
                        </ul>
                    </span>
                : null}
            </div>
        );
    }
};

export default PostsList;