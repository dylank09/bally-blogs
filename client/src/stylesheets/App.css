

div {
    max-width: 960px;
    margin: auto;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

nav {
    margin: 10px;
    border-radius: 5%;
}

.nav-link {
    margin: auto;
    border-radius: 15%;
    display:block;
    max-width: 100px;
    margin-left:2px;;
    border:1px solid #000;
    justify-content: center;
    align-items: center;
}

a {
    display:inline;
	padding:3px;
	text-decoration:none;
	background-color:#fff;
	color:#009;
}

a:hover {
    background-color:#009;
	color:#fff;
}