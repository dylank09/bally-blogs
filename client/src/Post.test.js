import React from 'react';
import { render, cleanup  } from '@testing-library/react';
import { createMemoryHistory } from 'history'
import App from './App';
import PostsList from './posts/PostsList';
import { Router } from 'react-router';
import PostPage from './posts/PostPage';
 
afterEach(cleanup);
const history = createMemoryHistory()

it ("renders app", () => {
  const { asFragment } = render(<App/>);
  expect(asFragment()).toMatchSnapshot();
});

it ("renders post list", () => {
  const { asFragment } = render(
                          <Router history={history}>
                            <PostsList/>
                          </Router>);
  expect(asFragment()).toMatchSnapshot();
});

it ("renders post page", () => {
  const { asFragment } = render(
                          <Router history={history}>
                            <PostPage/>
                          </Router>);
  expect(asFragment()).toMatchSnapshot();
});

it ("renders post component", () => {
  const { asFragment } = render(
                          <Router history={history}>
                            <PostPage/>
                          </Router>);
  expect(asFragment()).toMatchSnapshot();
});