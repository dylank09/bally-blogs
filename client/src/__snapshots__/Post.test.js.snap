// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders app 1`] = `
<DocumentFragment>
  <div>
    <div>
      <nav
        class="navbar"
      >
        <span>
          <a
            class="nav-link"
            href="/login"
          >
            Log In
          </a>
          <br />
          <a
            class="nav-link"
            href="/signup"
          >
            Sign Up
          </a>
        </span>
        <br />
        <a
          class="nav-link"
          href="/posts"
        >
          Posts
        </a>
        <br />
      </nav>
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders post component 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <a
        class="nav-link"
        href="/posts"
      >
        Posts
      </a>
            
      <br />
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
                  
      <br />
    </nav>
    <div>
      <h2>
        <a
          class="post-title"
          href="/posts/undefined"
        />
      </h2>
      <p
        class="post-body"
      >
          
      </p>
      <p
        class="post-date"
      >
         Created on:  
      </p>
      <br />
    </div>
  </div>
</DocumentFragment>
`;

exports[`renders post list 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
       
      <br />
    </nav>
    <ul
      class="posts-list"
    />
  </div>
</DocumentFragment>
`;

exports[`renders post page 1`] = `
<DocumentFragment>
  <div>
    <nav>
      <a
        class="nav-link"
        href="/posts"
      >
        Posts
      </a>
            
      <br />
      <a
        class="nav-link"
        href="/"
      >
        Home
      </a>
                  
      <br />
    </nav>
    <div>
      <h2>
        <a
          class="post-title"
          href="/posts/undefined"
        />
      </h2>
      <p
        class="post-body"
      >
          
      </p>
      <p
        class="post-date"
      >
         Created on:  
      </p>
      <br />
    </div>
  </div>
</DocumentFragment>
`;
